src/server/scripts/Spells/spell_dk.cpp

```diff
@@ -948,9 +948,17 @@ class spell_dk_death_grip : public SpellScriptLoader
 
             void HandleDummy(SpellEffIndex /*effIndex*/)
             {
-                if (Unit* target = GetHitUnit())
-                    if (!target->HasAuraType(SPELL_AURA_DEFLECT_SPELLS)) // Deterrence
-                        target->CastSpell(GetExplTargetDest()->GetPosition(), GetEffectValue(), true);
+                Unit* target = GetHitUnit();
+                Unit* caster = GetCaster();
+
+                if (!target || target->HasAuraType(SPELL_AURA_DEFLECT_SPELLS)) // Deterrence
+                    return;
+
+                const SpellInfo* spellInfo = sSpellMgr->GetSpellInfo(47528); // Mind Freeze
```

Comment:
SpellInfo const* spellInfo, and magic number too.
Btw, it dont looks a  correct way to check imune o.o

Reply:
`SPELL_DK_MIND_FREEZE       = 47528,` needed in enum list

Reply:
I didn't found any generic way to check interrupt immunity in core.

Reply:
Not tested but maybe:

if (target->GetMechanicImmunityMask() & MECHANIC_INTERRUPT)